### ISSUE TITLE
ICE when missing $ in front of a fluent variable

### ISSUE THREAD

**Issue by oli-obk** [rust-lang/rust#103539]
I tried this code:

```rust
the_slug =
    foo bar {boo} beep bop
```

I expected to see this happen: an error telling me `{boo}` needs to be `{$boo}`

Instead, this happened:

```
thread 'rustc' panicked at 'identifier: "hir_analysis_const_impl_for_non_const_trait", attr: Some("suggestion"), args: FluentArgs([("trait_name", String("A"))]), errors: [ResolverError(Reference(Message { id: "trait_name", attribute: None }))]', compiler/rustc_errors/src/translation.rs:105:17
```

**Comment by davidtwco** [comment]
The issue here is that this isn't a Fluent error, we'd get a nice diagnostic if it was, it is still valid Fluent which refers to another slug, `boo`, which we only find out about when actually trying to do a translation. Everything is written so that if this mistake happens in a non-default locale, then we'll fallback to English and have a good user experience. But if it happens in the default locale, then we just expect tests to catch it. We could probably walk the AST of the parsed Fluent in the `fluent_messages` macro and try catch it then.

**Comment by oli-obk** [comment]
Yea... I think if we could just improve the ICE that would be sufficient.
